File: app/src/main/java/com/lxf/tools/ApiService.java
package com.lxf.tools;

import com.lxf.tools.bean.RequestModel;
import com.lxf.tools.bean.ResponseModel;
import com.lxf.tools.bean.ResultEntity;
import com.lxf.tools.bean.SignRequestBody;

import java.util.Map;

import io.reactivex.Observable;
import okhttp3.ResponseBody;
import retrofit2.Response;
import retrofit2.http.Body;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface ApiService {
    @POST("getdataserver")
    Observable<ResponseModel<String>>  getVersion(@Body RequestModel requestModel);
    @POST("postdataserver")
    Observable<ResponseModel> post(@Body RequestModel requestModel);



    @POST("/api/terminal/v1/merchant/newlogin")
    Observable<ResultEntity<Map<String, String>>> login(
            @Body SignRequestBody body
    );


    @POST("api/game/user/login")
//    Observable<ResponseBody> login(@Query("phone") String phone, @Query("password") String password);
    Observable<Response<ResponseBody>> login(@Body Map<String,String> params);
}

File: rx_retrofit/src/main/java/com/lxf/rxretrofit/api/ApiService.kt
package com.lxf.rxretrofit.api

import io.reactivex.Observable
import okhttp3.MultipartBody
import okhttp3.ResponseBody
import retrofit2.http.*
import java.util.*


internal interface ApiService {

    @Streaming
    @GET
    fun downloadFile(@Url fileUrl: String): Observable<ResponseBody>

    @Multipart
    @POST
    fun uploadFiles(
            @Url uploadUrl: String,
            @Part files: List<MultipartBody.Part>
    ): Observable<ResponseBody>
}